### PR TITLE
Add enabled to pipedProvider and provider config

### PR DESCRIPTION
**Pull request by sbender9**
This makes it easy to turn on/off parts of the server settings without having to maintain multiple settings files.

### CODE REVIEW

**Comment by tkurki** [comment]
The idea is great but would you reconsider the implementation?

If you convert the `xxxConfig.map`s to `.reduce` you can convert & filter at the same time and there is no need to return null and filter on that, when you never called the `createXX`.

```
created = configs.reduce( (result, config) => {
   if (config.enabled) {
      result.push(createFoo(config)
   }
  return result
}, [])
```

**Comment by sbender9** [comment]
Done. Why does "if ( config.enable )" work when enable is not defined? I would expect that to be false.

**Comment by tkurki** [comment]
Does it? I think it is my mistake in the example i wrote on the commute...

**Comment by sbender9** [comment]
Let me test again to double check...

**Comment by sbender9** [comment]
Yeah, I was testing the wrong version. Fixed.

File: lib/pipedproviders.js [hunk 1]
```diff
@@ -54,6 +59,11 @@ module.exports = function(app) {
   function createPipeElement(elementConfig) {
     var options = elementConfig.options || {};
     options.app = app;
+    
+    if (typeof elementConfig.enabled != 'undefined'
+        && elementConfig.enabled == false)
+      return null
```
Comment (tkurki) [inline lib/pipedproviders.js]: Is this still needed?

Comment (sbender9) [reply lib/pipedproviders.js]: No, forgot to remove. Will remove when I get home...

**Comment by sbender9** [comment]
This should be good to go now

**Pull request merged**